common/thorhelper/thorsoapcall.cpp

```diff
@@ -2523,18 +2545,23 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                             idx = 0;
                         if (idx==startidx)
                         {
+                            clientSpan->recordException(e);
                             StringBuffer s;
                             master->logctx.CTXLOG("Exception %s", e->errorMessage(s).str());
                             processException(url, inputRows, e);
                             return;
                         }
                     } while (blacklist->blacklisted(url.port, url.host));
+
+                    //else recordException?
+                    clientSpan->recordException(e);
                 }
             }
             try
             {
                 checkTimeLimitExceeded(&remainingMS);
                 checkRoxieAbortMonitor(master->roxieAbortMonitor);
+                //per spec an http client span should be created here
```

Comment:
This comment poses the question, "why isn't it?"  Should the comment be deleted?

Reply:
To avoid being pedantic, but I feel it should be added.
I'm thinking the existing span would change to an internal and this http span would be a client span, child of the internal.